Disable AES-CFB.

nmahendru opened the pull request:
#### 572fec1979ec18aa50fda2389fc9768459851535
<pre>
Disable AES-CFB.
<a href="https://bugs.webkit.org/show_bug.cgi?id=266990">https://bugs.webkit.org/show_bug.cgi?id=266990</a>
radar://120000331

Reviewed by Pascoe.

It&apos;s been removed from the standard since 2016.
This change will disable AES-CFB by default.
Following the same pattern as:
<a href="https://github.com/WebKit/WebKit/pull/8392">https://github.com/WebKit/WebKit/pull/8392</a>

* LayoutTests/crypto/subtle/aes-cbc-cfb-decrypt-malformed-parameters.html:
* LayoutTests/crypto/subtle/aes-cbc-cfb-encrypt-malformed-parameters.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-export-key-jwk-length-128.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-export-key-jwk-length-192.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-export-key-jwk-length-256.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-export-raw-key.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-key-encrypt-decrypt.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-key.html:
* LayoutTests/crypto/subtle/aes-cfb-import-jwk-key-length-128.html:
* LayoutTests/crypto/subtle/aes-cfb-import-jwk-key-length-192.html:
* LayoutTests/crypto/subtle/aes-cfb-import-jwk-key-length-256.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-decrypt.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-encrypt.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-unwrap-jwk-key.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-unwrap-raw-key.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-wrap-jwk-key.html:
* LayoutTests/crypto/subtle/aes-cfb-import-raw-key.html:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::isAESCFBWebCryptoDeprecated):
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):

Canonical link: <a href="https://commits.webkit.org/272615@main">https://commits.webkit.org/272615@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a9b27a2e13e812ac10cf05d6fb9a36bc7b4595ff

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32311 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11048 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34132 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34865 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29254 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33161 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13399 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8254 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28809 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32729 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9314 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28874 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8111 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8249 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28805 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36206 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/27756 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29376 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29236 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34379 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/32401 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8380 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6322 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32241 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10035 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/28579 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/38824 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7544 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9011 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8212 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8940 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pascoej commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2216,6 +2216,20 @@ DefaultTextEncodingName:
+  status: stable
+  category: dom
+  humanReadableName: "Deprecate AES-CFB Web Crypto"
+  humanReadableDescription: "Enable DepreAES-CFB Web Crypto"
```
"Enable DepreAES-CFB" typo

pascoej reviewed (changes requested):
Please squash into one commit before this can land (One commit per final-version of a PR is standard.)

nmahendru committed:
Disable AES-CFB.
https://bugs.webkit.org/show_bug.cgi?id=266990
radar://120000331

Reviewed by Pascoe.

It's been removed from the standard since 2016.
This change will disable AES-CFB by default.
Following the same pattern as:
https://github.com/WebKit/WebKit/pull/8392

* LayoutTests/crypto/subtle/aes-cbc-cfb-decrypt-malformed-parameters.html:
* LayoutTests/crypto/subtle/aes-cbc-cfb-encrypt-malformed-parameters.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-export-key-jwk-length-128.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-export-key-jwk-length-192.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-export-key-jwk-length-256.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-export-raw-key.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-key-encrypt-decrypt.html:
* LayoutTests/crypto/subtle/aes-cfb-generate-key.html:
* LayoutTests/crypto/subtle/aes-cfb-import-jwk-key-length-128.html:
* LayoutTests/crypto/subtle/aes-cfb-import-jwk-key-length-192.html:
* LayoutTests/crypto/subtle/aes-cfb-import-jwk-key-length-256.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-decrypt.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-encrypt.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-unwrap-jwk-key.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-unwrap-raw-key.html:
* LayoutTests/crypto/subtle/aes-cfb-import-key-wrap-jwk-key.html:
* LayoutTests/crypto/subtle/aes-cfb-import-raw-key.html:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::isAESCFBWebCryptoDeprecated):
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):

Canonical link: https://commits.webkit.org/272615@main
